Frontend/ui-customer/src/Screens/CustomerUtils/AddNewAddressScreen/Controller.purs

```diff
@@ -248,30 +248,28 @@ eval (TagSelected index) state = do
     else
       if (validTag state.data.savedTags activeTag state.data.placeName) then
         continue state{ data  { activeIndex = Just index
-                              , selectedTag = getTag index}
-                      , props { isBtnActive = if (index == 2 && state.data.addressSavedAs /= "") || (index == 2 && state.props.editLocation == true && state.data.placeName /="" )
-                                                then (not state.props.placeNameExists && true)
-                                                else if (index == 1 || index == 0)
-                                                  then (not state.props.placeNameExists && true)
-                                                    else false
-                                                    }}
+                              , selectedTag = getTag index
+                              , addressSavedAs = if index /= 2 then "" else state.data.addressSavedAs}
+                      , props { placeNameExists = if index /= 2 then false else state.props.placeNameExists 
+                              , isBtnActive = (index == 2 && state.data.addressSavedAs /= "") || (index == 2 && state.props.editLocation == true && state.data.placeName /="" ) || index == 1 || index == 0
```

Comment:
can we not dependent on index to know which `Tag` is selected instead we can pass data type for tag to handle cases

Reply:
this way it would be easier to debug or add multiple tags which are yet to come